core/src/main/scala/org/apache/spark/util/logging/FileAppender.scala

```diff
@@ -58,20 +58,29 @@ private[spark] class FileAppender(inputStream: InputStream, file: File, bufferSi
   protected def appendStreamToFile() {
     try {
       logDebug("Started appending thread")
-      openFile()
-      val buf = new Array[Byte](bufferSize)
-      var n = 0
-      while (!markedForStop && n != -1) {
-        n = inputStream.read(buf)
-        if (n != -1) {
-          appendToFile(buf, n)
+      Utils.tryWithSafeFinally {
+        openFile()
+        val buf = new Array[Byte](bufferSize)
+        var n = 0
+        while (!markedForStop && n != -1) {
+          try {
+            n = inputStream.read(buf)
+          } catch {
+            case e: IOException =>
```

Comment:
I don't think it's worth changing as this is fine, but I think you could just write
```
case _: IOException if markedForStop => // do nothing because ...
```
... but maybe I'm missing something about how that works.